### PR TITLE
Feat/cip30 collateral

### PR DESCRIPTION
**Pull request by mchappell**
# Checklist

- [x] JIRA - [LW-5232](https://input-output.atlassian.net/browse/LW-5232)
- [x] Screenshots added.

---

## Proposed solution

Allow `api.getCollateral()` to round-trip within lace and provide set collateral to the DApp without forcing the user through the settings flow

## Testing

Notes added to ticket

## Screenshots

Screenshot 2023-09-04 at 13 08 49

[LW-5232]: https://input-output.atlassian.net/browse/LW-5232?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1906/6084242540/index.html) for [d8f18214](https://github.com/input-output-hk/lace/pull/484/commits/d8f182141ab16d7c331f9ec3a51b72bfc80f447f)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 27     | 8      | 0       | 0     | 35    | ❌     |

### CODE REVIEW

File: apps/browser-extension-wallet/src/features/dapp/components/collateral/CollateralContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      )
+        .getCollateralRequest()
+        .then(({ dappInfo: requestDappInfo, collateralRequest }) => {
+          // Set the summary information needed to display
+          setDappInfo(requestDappInfo);
+          // Determine if collateral can be set without further fragmentation of wallet UTxOs
+          let totalCoins = BigInt(0);
+          const usableUtxos: Wallet.Cardano.Utxo[] = [];
+          for (const utxo of collateralRequest?.utxos) {
+            const { coins } = utxo[1].value;
```
Comment (lgobbi-atix) [inline apps/browser-extension-wallet/src/features/dapp/components/collateral/CollateralContainer.tsx:122]: Is it okay to use the UTxO if it has assets?

Comment (mchappell) [reply apps/browser-extension-wallet/src/features/dapp/components/collateral/CollateralContainer.tsx:122]: The SDK only sends UTxOs which don't have assets to choose from. I expect to refactor this in future to make that the responsibility of the wallet client, not the SDK so we reduce sending them 'over the wire'

File: apps/browser-extension-wallet/src/features/dapp/components/collateral/CollateralContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  const [lockableUtxos, setLockableUtxos] = useState<Wallet.Cardano.Utxo[]>([]);
+  const redirectToCreateSuccess = useRedirection(dAppRoutePaths.dappTxSignSuccess);
+  const redirectToCreateFailure = useRedirection(dAppRoutePaths.dappTxSignFailure);
+
+  const confirmCollateral = useCallback(
+    async (utxos: Wallet.Cardano.Utxo[]) => {
+      try {
+        await inMemoryWallet.utxo.setUnspendable(utxos).catch((error) => {
+          console.log(error);
+        });
```
[the remark this answers is not in the document]
Comment (mchappell) [reply apps/browser-extension-wallet/src/features/dapp/components/collateral/CollateralContainer.tsx]: Amended in [17e8494937439567855d616c2d1547a1cd064a9c](https://github.com/input-output-hk/lace/pull/484/commits/17e8494937439567855d616c2d1547a1cd064a9c)

**Review by lgobbi-atix — APPROVED** [review]

**Review by mirceahasegan — APPROVED** [review]
Great work @mchappell ! 🥇

**Pull request merged**